### ISSUE TITLE
CLI args formating

### ISSUE THREAD

**Issue by ambsw-technology** [merll/docker-map#13]
This is a (sanitized, and wrapped for readability) [create](https://docs.docker.com/engine/reference/commandline/create/) command generated by `client.cli.get_cmd`:

```
 docker create 
--name="graylog_map.graylog" 
--links="(u'graylog_map.mongo', 'mongo')" 
--links="(u'graylog_map.elasticsearch', 'elasticsearch')" 
--hostname="graylog_map.graylog" 
--env="GRAYLOG_PASSWORD_SECRET=<pepper>" 
--env="GRAYLOG_ROOT_PASSWORD_SHA2=<password>" 
--env="GRAYLOG_REST_TRANSPORT_URI=https://<url>:12900" 
--volume="/data/graylog/data:/usr/share/graylog/data/journal:rw" 
--volume="/data/graylog/config:/usr/share/graylog/data/config:rw" 
graylog2/server:2.0.3-2
```

Here are the issues (and resolutions) I've identified so far:

 - `links` is actually `link`
     - in `client.cli.KWARGS_MAP` add `'links': 'link'`
 - the value for `link` is not collapsed correctly and should be `<name>:<name>`
     - The code block below resolves a couple issues: recursive calls for e.g. a list of tuples and cases where the kw (i.e. the outer flags) are None

```
def _transform_kwargs(ka):
    if ka is None:
        yield
    for key, value in iteritems(ka):
        cmd_arg = KWARG_MAP.get(key, key.replace('_', '-'))
        if value is None:
            continue
        for response in _transform_kwarg(cmd_arg, value):
            yield response


def _transform_kwarg(cmd_arg, value):
    if isinstance(value, list):
        for vi in value:
            for response in _transform_kwarg(cmd_arg, vi):
                yield response
    elif isinstance(value, dict):
        for ki, vi in iteritems(value):
            yield _mapping_format(cmd_arg, ki, vi)
    elif isinstance(value, tuple):
        yield _mapping_format(cmd_arg, value[0], value[1])
    elif isinstance(value, bool):
        yield _arg_format(cmd_arg, 'true' if value else 'false')
    else:
        yield _quoted_arg_format(cmd_arg, value)
```
 - `graylog_map.graylog` is not a valid hostname.  As mentioned [here](https://github.com/docker/compose/issues/229) valid hostnames need to be sanitized to `[a-zA-Z0-9\-]+` or left out.  Replacing non-alphanumeric characters with hyphens could be a valid option.  For my testing, I used a code block to skip it.

**Comment by merll** [comment]
Thanks for the input. This will be fixed in the next release.

**Comment by ambsw-technology** [comment]
As I documented in merll/docker-fabric#14, the hostname issue is due to a change in docker and not due to anything in your packages.

**Comment by merll** [comment]
Fixed in Docker-Map release `0.7.1`.
